### PR TITLE
Adds GanttChart component.

### PR DESCRIPTION
**Pull request by vijayjangid**
Screenshot 2024-02-09 at 8 56 20 AM

**Pending items:**
1. Understand the best way to author the parent/child relationship and dependencies between tasks.
2. Theme colors to be used for tasks (currently using the default palette coming from gantt-chart-react library.

**Known limitations:**
1. The task list (first three columns in table - Name, start and end date) are fixed. It can be just toggled but cannot be altered for a display or switching to another prop.
2. Although there are tabstops on tasks but click is not triggering.
3. In general, the library seems to be offering very minimal support for extension (in terms of UI). Even the css classes are generated using StyledComponents so overriding that is difficult as well.

@ricmars let's collaborate on this to understand if this fits our MVP.

### CODE REVIEW

**Comment by vijayjangid** [comment]
Updated the PR with following features:
1. Constellation design system theme compatible
2. Added details view card on item click
3. Code and Docs refactoring
theme and popover

**Pending items:**
UI Component:
1. Update item on drag events
2. Create menu to create Project, Task or Milestone on the fly

**Pega case and data:**
1. Figure out the data association between Project, Tasks and Milestones
2. Support for dependencies (multi select)
3. Type is not rendering in Details View

**Docs:**
1. Once the above two are done, take appropriate screenshots and udpate in docs

**Comment by vijayjangid** [comment]
Updated the Docs with relevant screenshots from App. This PR is now good to start review. 
@ricmars

**Comment by ricmars** [comment]
changes looks good - had a couple of issues
- drag-drop is not persisted on the client -the task goes back to the previous position in storybook
- the ID in the card should use a link with the preview button instead of the outside link (previewable)
image
-why have showDetailsColumns setting if there is a toggle on the UI already? Is it more the 'Default' setting for the show/hide column (similar to defaultViewMode) or is it more to hide this functionality on the UI?
- the dates are not localized and always show in English - that would be a problem

There are challenges with the 3rd party lib around truncation of the table and the tooltip display on top of the ui instead of being below the mouse which makes it hard to know what has been selected

Otherwise look very great!

**Comment by vijayjangid** [comment]
Thanks for the review @ricmars
1. Drag drop persist - on update of an item, I am reloading the details for entire chart to sort them accoringly. May be for storybook we can store it in WebStorage to mock that behavior.
2. ID as link button: Sure, that can be fixed. In real product, that's anyway come from Details view.
3. `showDetailsColumns` is default selection only. **Let me know if you want me to remove it.**
4. Dates are **a real challenge** as the library handles the rendering. We can only rely on Details view to show localized information.
5. True, I tried my best to facelift it while fixing the UX challenges. But it offers very little control to handle rendering.

File: src/components/Pega_Extensions_GanttChart/demo.stories.tsx [hunk 1]
```diff
@@ -160,17 +172,22 @@ const setPCore = () => {
             {
               name: 'ID',
               value: (
-                <Link href='https://www.pega.com' target='_blank'>
-                  {id}
-                </Link>
+                <h3>
+                  <Link href='/' previewable>
```
Comment (vijayjangid) [inline src/components/Pega_Extensions_GanttChart/demo.stories.tsx:176]: Adds previewable link

File: src/components/Pega_Extensions_GanttChart/demo.stories.tsx [hunk 2]
```diff
@@ -239,7 +256,24 @@ const setPCore = () => {
...
           });
         },
-        updateCaseEditFieldsData: () => {
+        updateCaseEditFieldsData: (
+          pzInsKey: string,
+          payload: any,
+          etag: string,
+          context: string
+        ) => {
+          let tasksFromWebStorage = getDataFromStorage();
```
Comment (vijayjangid) [inline src/components/Pega_Extensions_GanttChart/demo.stories.tsx:265]: stores updated item to local storage

File: src/components/Pega_Extensions_GanttChart/index.tsx [hunk 3]
```diff
@@ -189,9 +189,9 @@ export default function PegaExtensionsGanttChart(props: GanttChartProps) {
           getPConnect,
           item: task,
           updatedFieldValueList: {
-            Progress: gtrTask.progress,
-            Start: gtrTask.start,
-            End: gtrTask.end
+            [progressFieldName]: gtrTask.progress,
```
Comment (vijayjangid) [inline src/components/Pega_Extensions_GanttChart/index.tsx:192]: fixes an issue of not updating start/end dates

**Review by vijayjangid — COMMENTED** [review]
Done the requested changes.

**Pull request merged**